cockatrice: 2017-08-31 -> 2019-08-31

evanjs opened the pull request:
<!-- Nixpkgs has a lot of new incoming Pull Requests, but not enough people to review this constant stream. Even if you aren't a committer, we would appreciate reviews of other PRs, especially simple ones like package updates. Just testing the relevant package/service and leaving a comment saying what you tested, how you tested it and whether it worked would be great. List of open PRs: <https://github.com/NixOS/nixpkgs/pulls>, for more about reviewing contributions: <https://hydra.nixos.org/job/nixpkgs/trunk/manual/latest/download/1/nixpkgs/manual.html#chap-reviewing-contributions>. Reviewing isn't mandatory, but it would help out a lot and reduce the average time-to-merge for all of us. Thanks a lot if you do! -->
###### Motivation for this change
Somebody expressed some interest for this on IRC but couldn't get the package working.
Ended up basically updating the whole thing so why not make a PR.

Figured I'd use the latest stable version, as the newest beta is only a month ahead of it.

App runs and initializes card database and etc fine, and I can browse through the cards after that.
###### Things done
- formatting
- add `wrapQtAppsHook`
- use `fetchFromGitHub` instead of `fetchurl`
- don't construct name manually
- add homepage to meta
- remove repositories.get from meta
- use `mkDerivation` instead of `stdenv.mkDerivation` (per the Qt section of the [manual](https://nixos.org/nixpkgs/manual/#qt-default-nix-co-2))
___

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

spencerjanssen commented:
Thank you for the patch.  The changes look good to me except I get a 404 trying https://github.com/Cockatrice/Cockatrice/archive/2019-08-31.tar.gz, looks like the rev isn't correct?

I no longer use this package, if you or anyone else would like to take over maintainership please feel free.

evanjs committed:
cockatrice: 2017-08-31 -> 2019-08-31

- formatting
- add wrapQtAppsHook
- use fetchFromGitHub instead of fetchurl
- don't construct name manually
- add homepage to meta
- remove repositories.get from meta
- use mkDerivation instead of stdenv.mkDerivation
- add qtwebsockets dependency

evanjs commented:
Thanks for checking that.
Must have been after I made the change from `fetchurl` to `fetchGitHub`.
The only thing that bothers me at this point is the version name, but meh.

Anybody interested in maintaining this for now?
Totally unfamiliar with any policy related to lack of maintainers and keeping things on nixpkgs.

evanjs committed:
cockatrice: remove spencerjanssen from maintainers

evanjs committed:
cockatrice: add evanjs to maintainers
